### PR TITLE
Keep it snipple snupple

### PR DESCRIPTION
**Pull request by comradekingu**
Nearly nobody wants to read a CoC, and those that already have don't want to read it again to watch for invariant sections. If people that are problematic enough to be antisocial do happen to read the CoC, their nature or own judgement in being antisocial takes precedence over disobeying a text document.

The interesting part is the point of contact. I missed it because this CoC changed (again) and it used to be at the bottom somewhere. The Matrix link didn't work for me, but I am not registered.

### CODE REVIEW

**Comment by machiav3lli** [comment]
Sorry but I fully disagree, the importance of explicit and inclusive CoC for me far more important than the readability. Otherwise I would argue that CCCoC 2.0 far more readable and clear than its counterparts. Thanks.

**Comment by comradekingu** [comment]
It is neither explicit nor inclusive. Instead as vague as could be, and fundamentally geared against human interaction by replacing with an equally vague "professionalism". It is a political tool (as stated by its creator) that does nothing well, and can be removed in its entirety for huge communities (like on Weblate). When removed on Remmina, even donations went up. What matters is what you do, not what you put in a text document (and both those are not even good).

Actually the person writing it did so as being in charge of inclusivity at GitHub, and got fired later for not being inclusive. As you can see someone put a thumbs down on my initiative, even though the CoC says:
"* Giving and gracefully accepting constructive feedback", an irony of sorts.

You probably don't have the time to argue this, and it is your decision, as it is my decision to not contribute to projects and organisations that have a CoC. I have a full write-up of what I think about the covenant CoC if you are interested. I'll keep the nb_NO translations updated. I'd rather be banned than agree to that document, if you feel it is important that contributors pledge to honour it like it says. I didn't even know it was there, that is how much a CoC (at best) does to spread false security.

**Comment by Tiefkuehlpizze** [comment]
I like your write up and I just want to add, that I share the spirit. I can accept the existence of this document.

File: COC.md [hunk 1]
```diff
@@ -1,80 +0,0 @@
...
-
-* The use of sexualized language or imagery, and sexual attention or advances of any kind
-* Trolling, insulting or derogatory comments, and personal or political attacks
-* Public or private harassment
-* Publishing others' private information, such as a physical or email address, without their explicit permission
-* Other conduct which could reasonably be considered inappropriate in a professional setting
-
-## Enforcement Responsibilities
-
-Community leaders are responsible for clarifying and enforcing our standards of acceptable behavior and will take appropriate and fair corrective action in response to any behavior that they deem inappropriate, threatening, offensive, or harmful.
```
Comment (comradekingu) [inline COC.md:29]: Invariant section that is now in the hands of "community leaders" instead of "other members of the project's leadership" in the old version.

**Comment by comradekingu** [comment]
@Tiefkuehlpizze 
That is a correct assessment, but it is sadly not fully true (even in its ramifications).
Found the write-up again https://github.com/WeblateOrg/weblate/pull/2257/files
I added a part about "other members of the project's leadership" https://github.com/machiav3lli/oandbackupx/pull/115/files#r487583287 and pointed out what that is now. The 2.0 version is way worse.

**Comment by machiav3lli** [comment]
One of the most aspects of the inclusiveness of this document, the direct pointing at unacceptable behaviors or stances e.g. sexism, racism and harassment.

As a pedagogical, I understand your point. Keeping it simple and short gives it more reachability (more possibility that it'd be fully read), but on the same matter it doesn't put clear structure on how to (re)act on abusive/unacceptable actions. Keeping it short leaves a lot of freedom to the "community leaders"* to dictate what they want with no clear rules. With a CoC everyone has the right to question the decisions of the "community leaders" and this keeps the "community leaders" more informed on how/when to react.

* I don't like this term, "community leaders", but still, I think it does its job a lil bit better than the alternatives I thought of for now.

Any improvments to the existing CoC are welcome as long as those keep the elements of clarity and inclusiveness.

**Comment by comradekingu** [comment]
The pointing is about as indirect as whatever interpretation you want to give it, plus some banality.
You can have it whatever way you like, that is the point. No need to state anything. You are the community leader, just as much as you don't need given rules for infractions. Much less infractions that are judgement-calls. In any event, that is pretending the actual attempt isn't poor, and not what is actually in there.

What does it mean _no_ this and that? In the legal framework, you will find harassment is already illegal for example. In this instance it is actually clear what it means. You can throw the word "inclusivity" around, but it just adds authority to the virtual virtue signalling. You will find

> We as members, contributors, and leaders

has ideas

> for our community

I liked it more when users were the community, with no difference made between them, and forks being the tool to settle irreconcilable differences by. Back when there weren't any artificial power structures and barriers to entry.

Treat users like idiots, and all you will have is idiot users. Or however the saying goes. The reason "unwritten" rules remain unwritten, is because there is no replacing human interaction with a text. That lowers everyone to the lowest of standards.

Here is one particular titbit: "sexual attention or advances of any kind". I'll bring sexual attention to anything or anyone I like, according to my freedom of expression, biology and tact. Good luck in trying to undo biology, as if the wishful self-congratulating fanfare couldn't get any worse…
It is a political document that now doesn't allow "political attacks". Thought for food right there.

Give me one line that isn't worthless, redundant, or completely out to lunch that isn't in my piece of text. Just _one_. Much like the mafia and catholic church CoCs came from, this document extends privilege to hidden structures, and even invokes the escalation to "god"-given rights in opportune moment of robbing everyone else of them.

It is precisely in keeping it long and vague one grants (imaginary one would imagine) "community leaders" every power imaginable to do whatever they will. The longer the garbled text, the unclearer the rule. I questioned the rule, got shot down, and left. No hard feelings there, just don't pretend you side with "inclusivity", "community standards" or the like.

You don't get to dislike the term "community leaders" any more than you dislike putting it on others. You took a _pledge_ to do so. At-least in keeping with a document that also says 

> pledge to act and interact

Community leaders not only have the freedom to do as they wish, they have the 
> responsibility
to do so…

You may be a benevolent dictator, and putting these particular stripes on the suit doesn't change anything
> A warning with consequences for continued behaviour. No interaction with the people involved, including unsolicited interaction with those enforcing the Code of Conduct

Just stop answering, and you have "All complaints will be reviewed and investigated promptly and fairly." covered.

"Community leaders are responsible for clarifying and enforcing our standards of acceptable behaviour and will take appropriate and fair corrective action in response to any behaviour that they deem inappropriate, threatening, offensive, or harmful."

"fair" and (not so importantly) "that" is fluff there. "Our" standards.

And where-from comes such a invocation?

> Community leaders have the right and responsibility

Might makes right. What a surprise.

> and will communicate reasons for moderation decisions when appropriate

But will anything happen in the matter, and is there any choice in doing so?

> Community leaders will follow these Community Impact Guidelines.

Even a moron like me is able to turn that against you at ease. I for one don't think of that as a good thing. To the contrary, look at the voluntary action of offering service of CONTACT.md should human interaction between involved individuals fail.

Yes. I engaged in "aggression toward or disparagement of classes of individuals". That is a **Community Impact** under "permanent ban".

Just to be clear, there will be no

> public apology

I fully intend to

> avoiding interactions in community spaces as well as external channels like social media

if I so wish. And for what, a document with no value, but huge detriment to community and interaction. 
The true mark of the "feature" creep of bad writing is the constant flow of new versions to "amend" unforeseen issues.

The person inventing it got fired from GitHub, the people that made the "community interaction guidelines" mostly got fired from Mozilla because it can't afford to keep them. It will kill Mozilla (in my opinion). Be a good person if you are a good person. The Covenant CoC has always been an irrational piece of text for irrational actors. In my view that is intentional, but it is a gift to irrational actors unintentionally if one entertains the idea it is not. In the purest form of meritocracy we have, people adamant at destroying it since it supposedly isn't fair to them or their ends created it and use it for that purpose.

**Comment by machiav3lli** [comment]
It's a political tool and it's good so. I don't wanna see racist, homophobe or inhuman interaction in the community. As I already said I don't agree fully with the concept of "community leaders" that's because of the hierarchy that it imposes, still I'd take it anyday, if it helps to avoid such behaviors and in my experience it does.

Maybe you should join the community and see the communication for yourself before being prejudgmental. Don't reflect the bad experiences people had before on each new experience. This only shows (in my opinion) an unquestioned view of humans as hopeless non-learning creatures and here I would strongly disagree.

I would repeat my invitation for you to work on improving the existing document and join our community (on TG or Element) to see how it works for yourself (hopefully without any of the perceptions, based on your opinion of the document).

**Comment by comradekingu** [comment]
By that logic you should put in a rule about murder, because who wants that?

If someone can be racist and homophobic, and seize their behaviour or what do I know, improve their thought process because a piece of text says so, there is no stopping it. I mean you have to break laws to do some of the things mentioned in the text, but by its own admission, it was a pledge nobody asked for, nor took, nor read that did it. I support your laudable views, but not the method with which you seek to implement it. However if anyone was anti-social, a racist or a homophobe, they could remove anyone for any reason they deemed fit. Why aren't there more nice guys in the catholic church or the mafia if they have had a code of conduct for so long?

I have full faith in you doing something about instances of misbehavior if you did, and I would think less of you if you needed _a text document_ to do so. If you were asked to recite what is in that document, what are you left with? That is the position from which the wishful thinking stems, and supposedly is enacted.
Would love to know what it actually does anywhere. If a religious piterson does something, does it disprove or prove the existence of internal beliefs in the manifestation of something that isn't the actual action?

As a political tool the covenant CoC is tragically bad, but enough so that it exposes the proponents of it. I can't think of anything with fewer credentials, nor anything that has so many terrible decisions to show for. Again, if there are many states, does it prove greatness?

I can't join the community, because I have to pledge allegiance to your authoritarian text document to do so.
From my point of view there is a fundamental lapse in reason as to what community even is. It is held over my head as much as remarkable unwritten rules are. I don't need any of that.
Professionalism over humanity doesn't appeal to me, because I am a human. You actually don't want me or any people of my kind at all. I am not on any social media, and I am meritocratic.

I have already done enough to serve me a permanent ban from all engagement with the community, as decided by others for what I deem arbitrary reasoning.

Insofar as I am part of the libre software community, we don't need it, and we never did. The people that wrote the document are actually proponents of non-libre software. And they call it "ethical". Terrible consequences and nothing but trouble there too. I reflect every experience of the the code of covenant code of conduct text on every use of it, in every version of it. It has done nothing, it will continue to do nothing, and it is badly written at that.

You are using my point actually. Saying to someone they shouldn't be this and that, when there is no evidence or history of it, is condescending. Using superlatives and words that sound nice in a piece of text doesn't make it so. The actual instrumentation of facilitating non-meritocratic control does actually work, and they got as much right, because that is what they wanted to do. And did.

In my opinion the few lines of text I added greatly improves on the position. Not only in its brevity, but also in clarity.

**Comment by machiav3lli** [comment]
I think you're just overemotionalising this whole thing, mixing it with your personal stand for individuals' personal freedom.

I can't really see why you should be banned, you're just critisising a document which in the theory is there to allow people to argue fruitfully and don't go overboard with their private standing points over what is allowed to exist or not. Critical thinking is essential to human's developement but at the same time criticising just for critising won't bring us nowhere.

Maybe I should just repeat this once more: I understood your perspective and agree with most of it. Still, I see such a document is essential for a newly grounded community while at the same time the alternative that you've suggested so far isn't sufficiant enough to replace its function in my humble opinion.

**Comment by comradekingu** [comment]
Still no emotion in it on my part. My "stand for individuals' personal freedom" extends to you too, as the position you don't like pointed out, or extend to anyone else.

Point of the matter is, it is meant to. That part got purposely removed, and changed. There is a group dynamic there, which not only limits individual expression, but harms community. As it has harmed others, with no evidence of it helping anything.

How it could be "essential", when this is a fork, atop many libraries, in a libre software community that does does without, is beyond me. In my view it isn't even marginally useful, and I'll take examples of how it somehow is. There is no "in theory" about the text. You get exactly what is in there, and what you especially don't get to do is make up ideas about it. If people did that, it would be even more pointless, and even more lost on people that could potentially benefit. Preaching to the choir as it where, only inaudibly.
I would never try to replace that function, more than you don't like it being there.

Who do you think your friends are? Those people? Do they have fruitful communities, or are they inserting their endless drama into this one? The one I presume you were also in beforehand? The one they are now trying to counteract by challenging even software freedom down to making non-free licenses. Take your pick.
OK, you got some emotion out of me.

Take the ban on "trolling", a clear invocation of how each and every one of the people amending it and writing new versions, didn't understand culture. The irony of how libre software licenses and copyleft is actually trolling copyright is lost on them. Jokes are essential. Essential even to navigating social hierarchy, and that is how people develop enough of an emotional register to be functional in the world. The libre software world has always been humour in precisely this way.

Professionalism OTOH is made for career development, it is a functional tool in exerting corporate control, and ensure productivity at some guaranteed level. Over my dead rotating body will that take over the libre software community.

Denying sexuality isn't exactly a cornerstone in human development… The function is the same as it was for the catholic church. Silence, secret government, control, and anti-humanism. It does absolutely nothing without means of authority, and those always fall. Your humble opinion is worth as much as anyone implementing this code of conduct or the catholic one, both of which has happened in software.
"Having" a CoC does nothing. Why do you think Mozilla require grant applicants to have plans of enforcement and staff to do so? Why do you think people join projects purposely with this position in mind? Remember meritocracy? Who do they get rid of first? In my humble opinion the rule of the "community leaders" as a social hierarchy is not a good replacement. Remember trust?

If you can't trust or govern or act in your community without a CoC, what do you have?
You have politics beyond control. That is a political attack on my part. Stick to your guns and ban me, or remove the piece of text saying you can do so permanently for any sort of public communication within the community for such a reason. I not only could, but should be. As it "will" be acted upon.

What is the function in writing you support? Forget my alternative, what does it do over absolutely nothing?
Cite the passage, or come up with some examples. Anything. If you want something with no hopes of everyone understanding it, try giving https://remmina.org/ecosystem/ a shake. Yeah, then what?
Why not write "inclusivity" and "peace" 7 times each and hope it is a magic spell? Text does not govern, and the one you pledge allegiance to doesn't even try to do it either… If you pretend hard enough, you will have to reconcile this in faith. The exact opposite of critical thought isn't mindlessness.

With a CoC you neither get, nor build community. It is as an aside, always implemented after the fact.
Functional Commissariat. Meritless bloat.

**Comment by machiav3lli** [comment]
Dramatizing the whole discussion doesn't bring us anywhere. I'm not here to undermine or present the human in a negative way, matter of the fact it's the oppisite. Refusing to accept/tolerate sexism and racism is one of cornerstones of fighting for a social community that refuses the essence of capitlistic way of thinking. CoC is here not to control anything but rather give the community basis for tolerant non-exclusive interaction, a basis for a humane (in positive understanding of the word) communication. Does the neo-liberal values provide positive humans with such support? Easy answer, no.

**Comment by comradekingu** [comment]
@machiav3lli It isn't a dramatization, it is a functional argument, presented in a constructive manner.
What is deeply authoritarian and condescending towards contributors, and a division between contributors and users, does in no way need to be authoritarian. In doing all of those, you are championing that it does what exactly?

You can refuse to tolerate sexism and racism and whatever else, the idea of putting it in a text-file does what? When it is redundant to law, does it somehow mean projects that don't have a text-file saying this and that are lawless? Are things outside of the text-file allowed? If a text-file is the basis for humane communication then something is deeply wrong.

Tolerance is the furthest thing from what is in that document, objectively speaking.
**It is anti-human, in favour of professionalism.
The idea that capitalist thinking somehow is against community and moreover that fighting racism and sexism is somehow a way to counter it you will have to demonstrate more clearly, to me that sounds made up.**

What the document actually says is:

>Other conduct which could reasonably be considered inappropriate in a professional setting

That is as business-minded and non-community as anything, which would lend credence to actually _being_ neo-liberal and in tune with a capitalist way of thinking.

**The document has a bunch of provisions for anti-community behaviour.
Instead of engaging in a community manner, meaning with the people involved, it actually prevents it, when it is needed the most**:

>A temporary ban from any sort of interaction or public communication with the community for a specified period of time. No public or private interaction with the people involved, including unsolicited interaction with those enforcing the Code of Conduct, is allowed during this period.

>A warning with consequences for continued behavior. No interaction with the people involved, including unsolicited interaction with those enforcing the Code of Conduct, for a specified period of time.

This one is for a single violation(!)

> **Consequence**: A permanent ban from any sort of public interaction within the community.

**Comment by machiav3lli** [comment]
Hey again, sorry that I didn't have time lately the to continue our interesting discussion here.
I think this arguing about the why and what could take forever, in the spirit of finding alternatives that could cover the common points of interests I stumbled upon [Rust CoC](https://www.rust-lang.org/policies/code-of-conduct) and found it more simple while keeping the points that I hold for important and I would like to hear your thoughts about it.

**Comment by comradekingu** [comment]
@machiav3lli It is based on it, and the node.js one, which aren't model societies, along with https://steveklabnik.com/writing/a-sad-day-for-rust https://www.reddit.com/r/rustjerk/comments/7oonhj/lesscensored_version_of_the_ashley_williams_thread/ and whatever else. The point remains that it hasn't done anything, isn't proposing a to do anything, isn't introduced, nor followed as such. If one does nothing, why do you think everyone should be committed (is what it says) to another one based on it?

If no extraordinary evidence exist to back up these extraordinary claims, I am going with a most ordinary dismissal.

Why should it do anything of value here, when it isn't doing that for the people that made it?


    We are committed to providing a friendly, safe and welcoming environment for all, regardless of level of experience, gender identity and expression, sexual orientation, disability, personal appearance, body size, race, ethnicity, age, religion, nationality, or other similar characteristic.
    Please avoid using overtly sexual aliases or other nicknames that might detract from a friendly, safe and welcoming environment for all.
    Please be kind and courteous. There’s no need to be mean or rude.
    Respect that people have differences of opinion and that every design or implementation choice carries a trade-off and numerous costs. There is seldom a right answer.
    Please keep unstructured critique to a minimum. If you have solid ideas you want to experiment with, make a fork and see how it works.
    We will exclude you from interaction if you insult, demean or harass anyone. That is not welcome behavior. We interpret the term “harassment” as including the definition in the Citizen Code of Conduct; if you have any lack of clarity about what might be included in that concept, please read their definition. In particular, we don’t tolerate behavior that excludes people in socially marginalized groups.
    Private harassment is also unacceptable. No matter who you are, if you feel you have been or are being harassed or made uncomfortable by a community member, please contact one of the channel ops or any of the Rust moderation team immediately. Whether you’re a regular contributor or a newcomer, we care about making this community a safe place for you and we’ve got your back.
    Likewise any spamming, trolling, flaming, baiting or other attention-stealing behavior is not welcome.

**Comment by comradekingu** [comment]
@machiav3lli I don't pledge allegiance to this CoC or the Rust one, and I never did. Ban me if you believe in it, but I can't imagine you do.

**Comment by machiav3lli** [comment]
@comradekingu I don't know, you seem like a nice guy, having an aim which you take seriously and protesting what you find wrong in a civilized way, the way you're doing isn't something to be censored. 

Without arguing and discussions democracy dies. I'd like to continue our discussion. But am not having that much time lately and am still here waiting for you to suggest an improved (or fixed) version of a COC, based on views shared in this thread. Thanks

**Comment by comradekingu** [comment]
@machiav3lli All of the functional parts are present in the PR (which this is).
The original project made due without one, just as this one did prior to three months ago.
Ever since, I have just wasted a bunch of time on it, and it doesn't do anything anywhere else either. (?)

**Comment by comradekingu** [comment]
@machiav3lli The views in this discussion were

>I like your write up and I just want to add, that I share the spirit.

About my writing.

>I accept the existence of this document because it's just verbosely writing obvious things.

Neither here nor there.

>the importance of explicit and inclusive CoC for me far more important than the readability.

I can't understand this as anything but virtue-signaling.

>It's a political tool and it's good so.

Why? Has it done anything besides create fiascos? Does it not waste a lot of time?

>Maybe you should join the community and see the communication for yourself before being prejudgmental.

I am in the community insofar as contributions go. I never pledged allegiance to that text-file, and I never will. There is already one for GitHub, but I imagine you already knew because you care. Or?

**Comment by machiav3lli** [comment]
@comradekingu yeah there's one for GH, but this one is pointed at both GH and the community groups (Element & TG).
I understand that you won't take no for an answer to this issue. And yeah you belogns to community not only through your great contribution to the translations, but also specificly here on this thread. As said before, I'd be open to another, shorter form of CoC. Just introduce me to possible ones or let's create a new one together.

It's easy to see the problematic of such objects in ghe news as the good sides never make it to the news. Fortunately, there's been no situation where we had to go back to the document to solve communication issues.

So, you're fighting to keep it snipple snupple, would you be ready to make compromises to make a change (which can be adapted to other projects as well)?

**Comment by comradekingu** [comment]
Element comes with its own CoC too. In fact I think it has the same one…
Going back to the text-document is as warrant-less as the document itself. There is no rationale for having it, it has no value, and there is no authority in instating it on behalf of others. It is a tautology backed by absolutely nothing. That I can disregard it completely as such and not get banned means it is as hollow as could possibly be.

Yes, I am 100% open to ideas, this is what I (un-ironically) proposed:

> Welcome to the OAndBackupX community

> Accept and offer criticism constructively. Let anyone have their desired privacy.
Settle differences within these boundaries.
Finding yourself unable to do so, contact Antonios Hazim (machiav3lli), the project maintainer on [Matrix](https://matrix.to/#/!PiXJUneYCnkWAjekqX:matrix.org?via=matrix.org) or [Telegram](https://t.me/OAndBackupX).

That qualifies for both "shorter" and "another". Expect no greater miracles, but if you read the actual content or between the lines there I don't think anyone would think twice about contributing.
I am oblivious to why my proposal doesn't work in the projects I contribute the most to, Weblate and Remmina. My irrelevancy in other venues spans quite a lot of projects, and I am not seeing the positives of the Covenant CoC anywhere.

If no text-document and only culture made this project, what other functions do you need? Why does it have to be a _code of conduct_? A means of contact isn't a code of conduct, and at best my proposal is an appeal to decency and the de-escalation of conflict.

In a professional capacity, I have worked for some of the most prominent powers of "neo-liberal" and "capitalist" thinking. They have the CoCs, the self-congratulatory drivel, the awards, the statistics, the what have you. Only it is better written and better calculated. The professionalism-above-all-else is profit-driven, only there is no profit to be derived from it here. It comes at the _cost_ of community. It is ulterior motive and smoke and mirrors amounting to a PR cost worth of honesty. I _really_ don't nor understand what is to be gained by forcing it into the libre software world.

**Comment by machiav3lli** [comment]
So here we go again. I've being doing some research about the CoCs lately and caught couple of interesting & simpler versions. I know that you are clearly against a CoC in all forms (at least, this is how it seems to me), but I'd like to hear what you have to say on [this](https://geekfeminism.wikia.org/wiki/Community_anti-harassment/Policy) (the short and the long one) and [this](https://www.python.org/community/diversity). Thanks.

**Comment by comradekingu** [comment]
@machiave3lli I mistakenly started working on the translation again, only to find this is still a problem.

Coming from the country the furthest ahead in all matters of measured equality, I don't see a point in feminism, and it is consequently also completely dead here. Unsurprisingly neither of them even agreed with each-other even.

Pretending like "diversity" amounts to non-immutable characteristics is not only an affront to the capabilities of man, but directly counterproductive to its pretended aim. What it does is single out people, for things beyond their control.
That is the opposite of normalization. It would otherwise (and does) not cause issue in this ecosystem, and much less in a system of adoptable anonymity, online.

Moreover, you don't combat the operating idea that ones race isn't irrelevant to ones ideas by championing race as a more valuable metric.
Guess how far you go with these people challenging this concept of what their false authority is built on?
That is how much diversity of thought is worth to them.

What these people want is to instrument their own agenda and authority in places it doesn't belong, by tarnishing meritocracy.
You will find the 2.0 version of the covenant CoC replaces "project maintainers" with "community leaders", adding further bloat and laughable notions at that. Not by mistake I might add, as that has been a stated goal all along.

You will also find the very license scheme devised by the CoC creator isn't even libre software.
Not by accident has every single venue of tarnishing or attacking the libre software ecosystem been explored.
Python and Debian have lost their leadership only to arrive at exactly the same conclusion Intel did, which is that "diversity spending" is a bunch of money spent for no good reason. Do you intend to spend more money than Intel, or create systems of support better than Norway?
Look to Mozilla to see a company tying itself to these ideas while taking every bit of salary to the bank as entire "diverse" departments and actual engineers are axed. What is the lesson to take from it?
It is as fraudulent as BLM, and these people know. Much like some geek feminism wiki writers seemingly know they are up against biology with only oppression to make amends.

Failing to notice patterns of human nature arriving at ultimate expression of difference in adopting choice and doing away with stern plight, you would do well to arrive at unfruitful.
This is why systems of oppression are attemptedly back in.

To see what patterns it follows, note

>‘Reverse’ -isms, including ‘reverse racism,’ ‘reverse sexism,’ and ‘cisphobia’

Curiously, these intellectual titans have also put

> Criticizing racist, sexist, cissexist, or otherwise oppressive behavior or assumptions

under "Harassment includes:"

I'll, as is plainly obvious, criticize oppressive behavior or assumptions all day every day,
Let me just take the opportunity to send you a hug before you make this assumption on part of this entire project by way of

>Physical contact and simulated physical contact (eg, textual descriptions like “*hug*” or “*backrub*”) without consent or after a request to stop.

If like the Python blurb it is all virtue-signalling, how much is the rest of it worth?
Alone that isn't simple, it i simply unproductive.

https://github.com/biocore/scikit-bio doesn't have a CoC. Bio-diversity is their game.

>We have created this diversity statement because we believe that a diverse Python community is stronger and more vibrant. A diverse community where people treat each other with respect has more potential contributors and more sources for ideas.

This is a blind tautology. By what measure is not doing more of it adding to said vibrancy? Who on earth is going to be respectful just because we assume such people read it? What is that system respectful to?

"The Python Software Foundation and the global Python community welcome and encourage participation by everyone" needs disclaimers and appendixes how? The implication doesn't hold, but that is another matter. It boils down to "Welcome", which you will find added in the proposal.

**Comment by machiav3lli** [comment]
There's a lot of subjectivity in what you just wrote. What worked (and how it worked) in Norway isn't a standard for all countries and institutions to follow. There's surely some to learn from it, but there's huge discripancies when talking about theory and reality.

The Norse countries still topping the global gender gap list has also to do with some other facts as just ideological, historical or procedural facts. For once, the popultion of the norse countries (denmark, sweden, norway, iceland, greenland and finland) together is still less than half of that of germany or france. Just looking at at the top 10, you can see that only germany and rwanda have a population density of over 100.

What you're suggesting is a logical fallacy considering there's far more to that than just procedural actions.

Am also no fan of identity politics and find some of the stuff that can be understood today under such corny, but there should be clear lines declaring that everyone has the right to be itself. In a world where some old generations feel to be lost because of the new dynamic roles and some regulations, it's just logical to speak up for everyone's right and not to allow for discriminatory ressentiments to slip through.

Having a CoC isn't a one-way regulation, reponsibility is the other side of this. So whenever a "community leader" does something wrong, this also hold true for him/her.

I understand you ideological motive, but it's a product of (an)other environment(s) as the ones I grew up in and the experiences I made.

**Comment by comradekingu** [comment]
You no more than I, can't apply logic to abstraction, but you are for what is there, working on a tangent here, and end up with a flawed analysis by your own standards. What I think given insight into what liberty does on a societal level has no bearing on what applying draconian fake-laws onto a healthy society.

Nevertheless, the reason you get differences after things seem to even out is precisely that of post post-modernity.
This trend is the same anywhere, and you need powerful motivators to undo it. If it was just down to adjusting for population size, or disparities in age (Rwanda vs Germany) that would be easy. Not in the least is Oslo as densely populated as German cities, but that is also beside the point. One determining factor is sacrificing oneself for ones country. With that being less of a calling and everyone able to go to university, you end up with disparities previously lessened, along gender lines.
This pattern is the same in Africa as it is in northern Europe.
If you remove societal pressures completely to the tune of allowing for the most prestigious occupation to be there for the taking, like the application of the medical field, still not a single thing about sexual selection changed. Despite every effort to pretend otherwise. Thus, we have gender roles, and we always will have.
Some of which could be done away with, but the premise remains that of expression in the driving force of biology.
Short of evolutionary pressures which are greater of course. The control groups are dictatorships attempting to eradicate and remodel social interaction. I hope we agree results have been less than satisfactory, and not short of any feasible pressure.

As the argument actually pertains to, dealing in similarities and actual power-structure of the current CoC:

>Focusing on what is best not just for us as individuals, but for the overall community

This sounds precisely like the type of collectivism that isn't based in voluntary and free interaction.

"Demonstrating empathy" it is no more than it is "diverse".
How much of
> healthy community

can you have with

>sexual attention or advances of any kind

banned? None. None whatsoever. It is authoritarian, it has never worked, and hopefully never will.

This CoC is as much of a one-way regulation as it could be. It neither takes into account who built the program originally, nor does it even do the due diligence of asking its current contributors, which are said to pledge allegiance to it.

> Community leaders are responsible for clarifying and enforcing our standards of acceptable behavior and will take appropriate and fair corrective action in response to any behavior that they deem …

> Community leaders have the right and responsibility to remove, edit, or reject comments, commits, code, wiki edits, issues

> and other contributions that are not aligned to this Code of Conduct,

In actual practice the product here is even taken to be above the individual, which is in turn beneath the collective.
The expression of intent is completely gone as it pertains to any expression of behavior. That is the complete limitation to applied empathy.

The law is above the land, and the interpretation of it is down to enforcers.

>and will communicate reasons for moderation decisions when appropriate.

that don't even need to notify anyone of why decisions are made.

>All complaints will be reviewed and investigated promptly and fairly.
>All community leaders are obligated to respect the privacy and security of the reporter of any incident.

Show trials, without the show. The only thing falling outside of "right, or responsibility" is the _obligation_ to keep facing ones accuser as far away from fair investigation and trial as doing away with intent.

>A public apology may be requested.

Borrowing heavily here from the worst dictatorship in human history.

> This includes avoiding interactions in community spaces as well as external channels like social media.

No freedom of association either. Either that, or it is social punishment. Either way it is chilling.

> "A violation through a single incident" → "No interaction with the people involved"

Complete avoidance of mitigation, and every venue of applying the completely arbitrary rule of the land wrong opened.

The lowest infraction is:

> or other behavior deemed unprofessional or unwelcome in the community

A sustained level of which can lead to a _permanent ban_. No recourse.
Not only is the rights of the individual sacrificed for the common good, it is also placed in the hand of a monopoly of total power.
This is a class purposely changed from "project maintainers" to "community leaders".

The pretty language is to you and as a great example of similar rule-by-text, as arbitrary as replacing it with openly and codified racist doctrine, which also bans its critique.
From the feminist wiki:

> Criticizing racist, sexist, cissexist, or otherwise oppressive behavior or assumptions

As a common passage, these people think they are as far above the law as 

>including unsolicited interaction with those enforcing the Code of Conduct.

The hardest possible ostrization, utilized by the hardest cults of the world. Evidently these people aren't building communities any more than they manage to diversify them, along their most un-diverse thinking. Pun intended.

Above you is the ability to speak freely with whom you want. Yielded power paraded by holders of the titles "community leader".
Not elected, not re-elected, just total hegemony of social interaction and as authoritarian a hierarchical structure could ever be.
That is not community, it is authoritarianism masked as an appeal to "professionalism".

>considered inappropriate in a professional setting
>behavior deemed unprofessional or unwelcome

"Responsible" for upholding the rules are project leaders, which means people without any merit in leading a project all of a sudden are thought to represent the will of the people too. The worst worm of feudalism.

Not even just unmeritocratic by nature, it is downright destructive:

>A permanent ban from any sort of public interaction within the community.

>aggression toward or disparagement of classes of individuals.

You correctly applied the only pronouns possible with him/her, and I imagine you know under your own set of rules, by the very people that made them, this invokes a permanent ban. Elevating hypocrisy to anything but the rife and ample examples of it isn't what I think you are trying to do here.

The ability to fork is the only thing they couldn't take, and predictably we in turn got a new line of licensing to take care of that.

TL;DR There is nothing about the nature of that document, its history, its legacy, or its motives that warrant a just regulatory structure, as opposed to a self-righteous and holy _total_ dictatorship.
